app/assets/javascripts/sufia/fixedsticky.js

```diff
@@ -0,0 +1,193 @@
+						[ !turnOn ? 'addClass' : 'removeClass' ]( S.classes.inactive );
+				},
+				viewportHeight = $( window ).height(),
+				position = $el.data( S.keys.position ),
+				skipSettingToFixed,
+				elTop,
+				elBottom,
+				$parent = $el.parent(),
+				parentOffset = $parent.offset().top,
+				parentHeight = $parent.outerHeight();
+
+			if( initialOffset === undefined ) {
+				initialOffset = $el.offset().top;
+				$el.data( S.keys.offset, initialOffset );
+				$el.after( $( '<div>' ).addClass( S.classes.clone ).height( height ) );
+			} else {
+				$el.next( '.' + S.classes.clone ).height( height );
+			}
+
+			if( !position ) {
+				// Some browsers require fixed/absolute to report accurate top/left values.
+				skipSettingToFixed = $el.css( 'top' ) !== 'auto' || $el.css( 'bottom' ) !== 'auto';
+
+				if( !skipSettingToFixed ) {
+					$el.css( 'position', 'fixed' );
+				}
+
+				position = {
+					top: $el.css( 'top' ) !== 'auto',
+					bottom: $el.css( 'bottom' ) !== 'auto'
+				};
+
+				if( !skipSettingToFixed ) {
+					$el.css( 'position', '' );
+				}
+
+				$el.data( S.keys.position, position );
+			}
+
+			function isFixedToTop() {
+				var offsetTop = scroll + elTop;
+
+				// Initial Offset Top
+				return initialOffset < offsetTop &&
+					// Container Bottom
+					offsetTop + height <= parentOffset + parentHeight;
+			}
+
+			function isFixedToBottom() {
+				// Initial Offset Top + Height
+				return initialOffset + ( height || 0 ) > scroll + viewportHeight - elBottom &&
+					// Container Top
+					scroll + viewportHeight - elBottom >= parentOffset + ( height || 0 );
+			}
+
+			elTop = getPx( $el.css( 'top' ) );
+			elBottom = getPx( $el.css( 'bottom' ) );
+
+			if( position.top && isFixedToTop() || position.bottom && isFixedToBottom() ) {
+				if( !isAlreadyOn ) {
+					toggle( true );
+				}
+			} else {
+				if( isAlreadyOn ) {
+					toggle( false );
+				}
+			}
+		},
+		destroy: function( el ) {
+			var $el = $( el );
+			if (S.bypass()) {
+				return $el;
+			}
+
+			return $el.each(function() {
+				var $this = $( this );
+				var id = $this.data( S.keys.id );
+				$( win ).unbind( '.fixedsticky' + id );
+
+				$this
+					.removeData( [ S.keys.offset, S.keys.position, S.keys.id ] )
+					.removeClass( S.classes.active )
+					.removeClass( S.classes.inactive )
+					.next( '.' + S.classes.clone ).remove();
+			});
+		},
+		init: function( el ) {
+			var $el = $( el );
+
+			if( S.bypass() ) {
+				return $el;
+			}
+
+			return $el.each(function() {
+				var _this = this;
+				var id = uniqueIdCounter++;
+				$( this ).data( S.keys.id, id );
+
+				$( win ).bind( 'scroll.fixedsticky' + id, function() {
+					S.update( _this );
+				}).trigger( 'scroll.fixedsticky' + id );
+
+				$( win ).bind( 'resize.fixedsticky' + id , function() {
+					if( $el.is( '.' + S.classes.active ) ) {
+						S.update( _this );
+					}
+				});
+			});
+		}
+	};
+
+	win.FixedSticky = S;
+
+	// Plugin
+	$.fn.fixedsticky = function( method ) {
```

Comment:
If this isn't code that you wrote, it should be in `vendor/assets/javascripts/fixedsticky.js`